include/internal/quic_types.h

```diff
@@ -27,7 +27,7 @@
 /* QUIC packet number spaces. */
 #  define QUIC_PN_SPACE_INITIAL           0
 #  define QUIC_PN_SPACE_HANDSHAKE         1
-#  define QUIC_PN_SPACE_APP               2
+#  define QUIC_PN_SPACE_APP               2     /* Must be last */
```

Comment:
(prior, but still...) Could this not be done with an enum?

Then it could be

```
enum {
    QUIC_PN_SPACE_INITIAL = 0,
    QUIC_PN_SPACE_HANDSHAKE,
    /* New entries must go here, so that QUIC_PN_SPACE_APP is last */
    QUIC_PN_SPACE_APP,
    QUIC_PN_SPACE_NUM
};
```

And if OpenSSL has `static_assert()` or some such, adding

```
OSSL_static_assert(QUIC_PN_SPACE_APP + 1 == QUIC_PN_SPACE_NUM);
```

would be useful

Reply:
Yes, it could.  The project isn't good at using enums.